internal/controller/kyma_controller.go

```diff
@@ -405,21 +406,73 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 		}
 
 		r.RemoteClientCache.Del(client.ObjectKeyFromObject(kyma))
-		if err := remote.RemoveFinalizerFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
+		if err := remote.RemoveFinalizersFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
 			err = fmt.Errorf("error while trying to remove finalizer from remote: %w", err)
 			r.enqueueWarningEvent(kyma, deletionError, err)
 			return ctrl.Result{}, err
 		}
 
-		logger.Info("removed remote finalizer")
+		logger.Info("removed remote finalizers")
+	}
+
+	manifestsExist, err := r.cleanupManifestCRs(ctx, kyma)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if manifestsExist {
+		return ctrl.Result{Requeue: true}, nil
 	}
 
 	r.Metrics.CleanupMetrics(kyma.Name)
 
 	controllerutil.RemoveFinalizer(kyma, shared.KymaFinalizer)
+
 	return ctrl.Result{Requeue: true}, r.updateKyma(ctx, kyma)
 }
 
+func (r *KymaReconciler) cleanupManifestCRs(ctx context.Context, kyma *v1beta2.Kyma) (bool, error) {
+	relatedManifests, err := r.getRelatedManifestCRs(ctx, kyma)
+	if err != nil {
+		err = fmt.Errorf("error while trying to get manifests: %w", err)
+		r.enqueueWarningEvent(kyma, deletionError, err)
```

Comment:
this `r.enqueueWarningEvent` can be extracted to upper layer

```
manifestsExist, err := r.cleanupManifestCRs(ctx, kyma)
if err != nil {
    r.enqueueWarningEvent
   return ctrl.Result{}, err
}
```

I'm fine with this return `(bool, error)`, however for this function, if you introduce here a customized error when manifest CR still exists, you just need to return error, and you don't need to deal with `err` and `manifestsExist` specifically, wdyt?
```
if r.relatedManifestCRsAreDeleted(relatedManifests) {
    return ErrManifestCRStillExists
}
```

Reply:
But this will not work in the case there is no error but we call the delete manifests. We want to requeue in this case so that we are sure that the manifests are deleted before we can remove the Kyma finalizer. If we just rely on the error, we can't do this

Reply:
sorry, I'm wrong, this `ErrManifestCRStillExists` is not returned in that condition, but in the end.
```
if r.relatedManifestCRsAreDeleted(relatedManifests) {
    return nil
}

if err = r.deleteManifests(ctx, relatedManifests); err != nil {
    return err
}
return ErrManifestCRStillExists
```